collectors/plugins.d/pluginsd_parser.c

```diff
@@ -124,7 +129,13 @@ PARSER_RC pluginsd_dimension_action(void *user, RRDSET *st, char *id, char *name
     UNUSED(user);
     UNUSED(algorithm);
 
-    RRDDIM *rd = rrddim_add(st, id, name, multiplier, divisor, algorithm_type);
+    uuid_t  *guid = &((PARSER_USER_OBJECT *)user)->guid;
+
+    RRDDIM *rd = rrddim_add_custom(
+        st, id, name, multiplier, divisor, algorithm_type, st->rrd_memory_mode, 0, uuid_is_null(*guid) ? NULL : guid);
+#ifdef ENABLE_DBENGINE
+    uuid_clear(((PARSER_USER_OBJECT *) user)->guid);
```

Comment:
Same as above, maybe make this conditional?

Reply:
Good idea.. I will see if uuid_is_null is cheap